system/metrics/sinks/file/fileSink.cpp

```diff
@@ -65,12 +65,39 @@ void FileMetricSink::writeMeasurementToFile(const std::shared_ptr<IMetric> &pMet
     }
 
     const char *unitsStr = pManager->queryUnitsString(pMetric->queryUnits());
-    if (unitsStr)
+
+    if (pMetric->queryMetricType() != METRICS_HISTOGRAM)
     {
-        name.append(".").append(unitsStr);
+        if (unitsStr)
+        {
+            name.append(".").append(unitsStr);
+        }
+        fprintf(fhandle, "%s -> %" I64F "d, %s\n", name.c_str(), pMetric->queryValue(), pMetric->queryDescription().c_str());
     }
+    else
+    {
+        std::vector<__uint64> values = pMetric->queryHistogramValues();
+        std::vector<__uint64> limits = pMetric->queryHistogramBucketLimits();
+        size_t numValues = values.size();
+        __uint64 cumulative;
+
+        for (int i=0; i<numValues-1; ++i)
+        {
+            cumulative += values[i];
+            fprintf(fhandle, "  name=%s, bucket le %" I64F "d=%" I64F "d\n", name.c_str(), limits[i], cumulative);
+        }
+
+        // inf
+        cumulative += values[numValues-1];
```

Comment:
what is the significance of the "inf" comment? The comment should be explicit and as informative as possible